File: ShadowFrameworkLite20/src/shadow/system/data/objects/SFIntArray.java
package shadow.system.data.objects;

import shadow.system.data.SFInputStream;
import shadow.system.data.SFOutputStream;

public class SFIntArray extends SFPrimitiveType{

	private int[] intValues;
	
	
	public SFIntArray(int intValuesSize) {
		super();
		this.intValues = new int[intValuesSize];
	}

	public int[] getIntValues() {
		return intValues;
	}
	
	

	public void setIntValues(int[] intValues) {
		this.intValues = intValues;
	}

	@Override
	public void readFromStream(SFInputStream stream) {
		int n=stream.readShort();
		intValues=stream.readInts(n);
	}
	
	@Override
	public void writeOnStream(SFOutputStream stream) {
		stream.writeShort((short)intValues.length);
		stream.writeInts(intValues);
	}

	@Override
	public SFIntArray clone(){
		return new SFIntArray(intValues.length);
	}
}

File: ShadowFrameworkJS/src/shadow/system/data/objects/SFIntArray.js
//Java to JS on 15/03/2012

function SFIntArray(n){
	this.intValues=new Array();
	//n is ignored
}

inherit(SFIntArray,SFPrimitiveType);

SFIntArray.prototype["getIntValues"]=function(){
			return this.intValues;
		};

SFIntArray.prototype["readFromStream"]=function(stream){
			var  n = stream.readInt();
			this.intValues  = stream.readInts(n);
		};

SFIntArray.prototype["writeOnStream"]=function(stream){
			stream.writeInt(this.intValues.length);
			stream.writeInts(this.intValues);
		};

SFIntArray.prototype["clone"]=function(){
			return new SFIntArray(intValues.length);
		};
